src/microbe_stage/CompoundCloudPlane.cs

```diff
@@ -335,7 +333,9 @@ public void ConvertToCloudLocal(Vector3 worldPosition, out int x, out int y)
             if (freeSpace < generousAmount)
                 continue;
 
-            float taken = slot.TakeCompound(localX, localY, fractionToTakeOnce) *
+            var fractionToTake = 1.0f - (float)Math.Pow(0.5f, delta / Constants.CLOUD_ABSORPTION_HALF_LIFE);
```

Comment:
Couldn't this line be at the top of the function to only run it once, instead of 4 times?